### PR TITLE
filter added to system/stream

### PR DESCRIPTION
**Pull request by curiosityfoundation**
am not sure where to add a test. based on system/stream/mapConcat

### CODE REVIEW

File: packages/system/src/Stream/Stream/filter.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import { mapChunks } from "./mapChunks"
+
+/**
+ * Applies the predicate to each element and allows passing elements 
+ * to reach the output of this stream.
+ */
+export const filter = <O>(f: Predicate<O>) => <R, E>(
+  self: Stream<R, E, O>
+): Stream<R, E, O> =>
+  pipe(
```
Comment (qlonik) [inline packages/system/src/Stream/Stream/filter.ts:14]: Looks like this pipe is not necessary.

File: packages/system/src/Stream/Stream/filter.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    self,
+    mapChunks((o) => Array_.chain_(o, (o) => (f(o) ? [o] : [])))
```
Comment (qlonik) [inline packages/system/src/Stream/Stream/filter.ts:16]: It seems that it is possible to use curried version `Array_.chain` instead of uncurried one.

**Review by qlonik — COMMENTED** [review]
Looks pretty good. You also need to reexport this method in one of the index files. Let's see what Michael says about it too.

File: packages/system/src/Stream/Stream/filter.ts (hunk 1 above)
Comment (curiosityfoundation) [reply packages/system/src/Stream/Stream/filter.ts:14]: happy to make this change - original thought was to model it after mapConcat which uses pipe and Array_.chain_

File: packages/system/src/Stream/Stream/filter.ts (hunk 2 above)
Comment (mikearnaldi) [reply packages/system/src/Stream/Stream/filter.ts:16]: we should use here the Array filter function as they do in zio:
https://github.com/zio/zio/blob/master/streams/shared/src/main/scala/zio/stream/ZStream.scala#L1245

**Comment by mikearnaldi** [comment]
@curiosityfoundation thanks, I will merge it for now and patch in master directly. We are trying to keep implementations in line with ZIO as much as possible so I will change A.chain to use the A.filter as it is in https://github.com/zio/zio/blob/master/streams/shared/src/main/scala/zio/stream/ZStream.scala#L1245

**Pull request merged**